DataFormats/Math/test/cudaAtan2Test.cu

```diff
@@ -0,0 +1,133 @@
+/**
+ * Derived from the nVIDIA CUDA 8.0 samples by
+ *
+ *   Eyal Rozenberg <E.Rozenberg@cwi.nl>
+ *
+ * The derivation is specifically permitted in the nVIDIA CUDA Samples EULA
+ * and the deriver is the owner of this code according to the EULA.
+ *
+ * Use this reasonably. If you want to discuss licensing formalities, please
+ * contact the author.
+ *
+ *  Modified by VinInn for testing math funcs
+ */
+
+/* to run test
+foreach f ( $CMSSW_BASE/test/$SCRAM_ARCH/DFM_Vector* )
+echo $f; $f
+end
+*/
+
+#include "cuda/api_wrappers.h"
+
+#include <iostream>
+#include <iomanip>
+#include <memory>
+#include <algorithm>
+#include <chrono>
+#include<random>
+
+#include<cassert>
+
+#include<DataFormats/Math/interface/approx_atan2.h>
```

Comment:
spaces and ```<...>``` for system includes vs ```"..."``` for CMSSW includes would be nice

Reply:
```#include <cassert>```
instead of ```#include<cassert>```

Reply:
tests can be updated in a follow up PR, which will not require large base recompilation.

If there is no plan to cleanup the copy-pasted ifdefs for binary unions, I'd rather we merge this already

Reply:
I just updated the copy-pasted ifdefs.
I was on the the todo list, never done because of the  large base recompilation.